test/createAzureResource.test.ts

```diff
@@ -59,26 +57,82 @@ suite('Create Azure Resources', async function (this: ISuiteCallbackContext): Pr
+        assert.equal(result, `Hello ${resourceName1}`, `The result should be "Hello ${resourceName1}" rather than ${result} and the triggerUrl is ${triggerUrl}`);
+    });
+
+    test('Deploy CSharp project (windows)', async () => {
+        const resourceName2: string = getRandomHexString().toLowerCase();
+        await clearProjectFile(projectPath);
+        const functionName: string = 'HttpTriggerCSharp';
+        resourceGroupsToDelete.push(resourceName2);
+        await runWithFuncSetting('projectLanguage', ProjectLanguage.CSharp, async () => {
+            await runWithFuncSetting('advancedCreation', undefined, async () => {
+                ext.ui = new TestUserInput([resourceName2]);
+                await vscode.commands.executeCommand('azureFunctions.createFunctionApp', testAccount.getSubscriptionId());
+                const createdApp: WebSiteManagementModels.Site = await webSiteClient.webApps.get(resourceName2, resourceName2);
+                assert.ok(createdApp);
+                await runWithFuncSetting('projectRuntime', ProjectRuntime.v2, async () => {
+                    const templateId: string = 'Azure.Function.CSharp.HttpTrigger.2.x';
+                    const nameSpace: string = 'Company.Function';
+                    const accessRights: string = 'function';
+                    await vscode.commands.executeCommand('azureFunctions.createNewProject', projectPath, ProjectLanguage.CSharp, ProjectRuntime.v2, false, templateId, functionName, { namespace: nameSpace, AccessRights: accessRights });
+                    await validateProject(projectPath, getCSharpValidateOptions('testOutput', 'netcoreapp2.1'));
+                });
+            });
+        });
+
+        ext.ui = new TestUserInput([testAccount.getSubscriptionId(), resourceName2, 'Update remote runtime', 'Deploy']);
+        await vscode.commands.executeCommand('azureFunctions.deploy');
+        await delay(500);
+
+        //Verify the deployment result through triggerUrl
+        const getFunction: WebSiteManagementModels.FunctionEnvelopeCollection = await webSiteClient.webApps.listFunctions(resourceName2, resourceName2);
+        assert.equal(getFunction[0].name, `${resourceName2}/${functionName}`);
+        const fullId: string = <string>(getFunction[0].id);
+        const triggerUrl: string | undefined = (<FunctionTreeItem>await ext.tree.findTreeItem(fullId)).triggerUrl;
+        const result: string = execSync(`curl ${triggerUrl}"&"name=${resourceName2}`).toString();
+        assert.equal(result, `Hello, ${resourceName2}`, `The result should be "Hello, ${resourceName2}" rather than ${result} and the triggerUrl is ${triggerUrl}`);
     });
 
     test('createFunctionApp (Advanced)', async () => {
-        const resourceName2: string = getRandomHexString();
+        const resourceName3: string = getRandomHexString();
         const resourceGroupName: string = getRandomHexString();
```

Comment:
It doesn't seem like there's a need to have resourceGroupName and resourceName3.  In this test, just replace resourceGroupName with resourceName3.  I'd also just make it all lowercase so you can use it for the storageAccountName.

Reply:
The previous code is the same as your idea, but Eric added a comment to "createFunctionApp (Advanced)", he said it's a little too similar to basic create. You can see https://github.com/microsoft/vscode-azurefunctions/pull/1128#discussion_r273752258